libnavui-status/src/main/java/com/mapbox/navigation/ui/status/model/StatusFactory.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.ui.status.model
+
+import androidx.annotation.DrawableRes
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+
+/**
+ * An object that creates new [Status] instances.
+ */
+@ExperimentalMapboxNavigationAPI
+object StatusFactory {
+
+    /**
+     * Instantiate a new status with a given field values.
+     *
+     * @param message The message field value.
+     * @param duration The duration field value in milliseconds. The default value is `0`.
+     * @param animated The animate field value. The default value is `true`.
+     * @param spinner The spinner field value. The default value is `false`.
+     * @param icon The icon field value. The default value is `0`.
+     */
+    @JvmStatic
```

Comment:
We can also add `JvmOverloads` annotation, so that Java developers benefit from default parameters as well.

Reply:
Great idea. I'll add it now.